### PR TITLE
Include readout fidelity in fidelity optimization

### PR DESCRIPTION
**Pull request by notmgsk**

### CODE REVIEW

File: tests/qpu-test-files/Aspen-4-2Q-B.qpu [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
```
Comment (erichulburd) [inline tests/qpu-test-files/Aspen-4-2Q-B.qpu:1]: readout-a.qpu replaces this, so this can be deleted.

**Review by erichulburd — CHANGES_REQUESTED** [review]

**Comment by stylewarning** [comment]
Can @ecpeterson look @ this?

File: src/addresser/addresser-common.lisp [hunk 2]
```diff
@@ -572,7 +572,6 @@ If DRY-RUN, this returns T as soon as it finds an instruction it can handle."
     (let ((locations (or locations qubit-cc)))
       (assert (not (apply-rewiring-l2p working-l2p logical)) (logical)
               "Qubit ~a already assigned" logical)
-
```
Comment (ecpeterson) [inline src/addresser/addresser-common.lisp:575]: daring

File: src/addresser/fidelity-addresser.lisp [hunk 3]
```diff
@@ -125,7 +126,17 @@
                (when any-assigned?
                  (a:when-let ((cost (2q-cost gate logical-qubits physical-qubits tier-index)))
                    (incf gate-count)
-                   (incf gate-weights-cost cost)))))))))
+                   (incf gate-weights-cost cost)))))))
+        (when (typep gate 'measurement)
```
Comment (ecpeterson) [inline src/addresser/fidelity-addresser.lisp:130]: yikes, i didn't know this was missing. this is a good fix.

**Review by ecpeterson — APPROVED** [review]

**Comment by notmgsk** [comment]
Merging.

**Pull request merged**